test/browser/browser_asrouter_cfr.js

```diff
@@ -372,3 +372,15 @@ add_task(async function test_onLocationChange_cb() {
 
   Assert.equal(count, 2, "We moved to a new document");
 });
+
+add_task(async function test_matchPattern() {
+  let count = 0;
+  const triggerHandler = () => ++count;
+  ASRouterTriggerListeners.get("frequentVisits").init(triggerHandler, [], ["*://*.example.com/"]);
```

Comment:
I don't think you need to re-test the whole set of functionality of `PatternMatch` here, but I'd like to see a couple more types of patterns